explorer/src/lib/dusk/string/decodeHexString.js

```diff
@@ -0,0 +1,31 @@
+import { hexToBytes, isValidHex } from "./";
+
+/**
+ * Decodes a hexadecimal string to utf-8 and formats it if it's a JSON string
+ *
+ * @param {string} value
+ * @returns {string}
+ */
+function decodeHexString(value) {
+  if (!isValidHex(value)) {
+    return value;
+  }
+
+  const bytes = hexToBytes(value);
+  const decoder = new TextDecoder();
+  const decodedString = decoder.decode(bytes);
```

Comment:
This could fail, if the byte sequence does not form a valid UTF-8 string.

We could wrap in try/catch block - if you want to return null in case of failure or throw the error is up to you.